docker/Dockerfile.backend

```diff
@@ -54,8 +54,10 @@ RUN chmod -R 744 $BUF_DIR \
 COPY requirements.txt .
 COPY requirements_dev.txt .
 
-RUN pip install --require-hashes --no-deps -r requirements.txt
+RUN pip install --require-hashes --no-deps -r requirements.txt -t /app/oppia/third_party/python_libs
 RUN pip install --require-hashes --no-deps -r requirements_dev.txt
+RUN pip download --no-deps -r requirements.txt -d /root/.cache
```

Comment:
Why are the package dependencies not being installed here?

Reply:
we need to pass the `--no-deps` flag because pip needs a pinned version for all deps, but some package deps might not be pinned to a specific version

please refer to an answer on [stackoverflow](https://stackoverflow.com/questions/75729749/getting-error-in-require-hashes-mode-all-requirements-must-have-their-versi)
![image](https://github.com/oppia/oppia/assets/91013793/8482a2df-dcef-4745-aaa8-422149e483eb)

Reply:
are the dependencies also mentioned in the requirements?  Since the packages need the dependencies to work.

Reply:
yes, the deps throwing errors are present in `requirements.txt`.